test/e2e/service.go

```diff
@@ -1848,20 +1863,24 @@ spec:
 		}
 		err = wait.PollImmediate(retryInterval, retryTimeout, checkNumberOfETPRules(1, fmt.Sprintf("[1:%d] -A OVN-KUBE-ETP", pktSize)))
 		framework.ExpectNoError(err, "Couldn't fetch the correct number of iptable rules, err: %v", err)
-		err = wait.PollImmediate(retryInterval, retryTimeout, checkNumberOfETPRules(1, fmt.Sprintf("[1:%d] -A OVN-KUBE-SNAT-MGMTPORT", pktSize)))
-		framework.ExpectNoError(err, "Couldn't fetch the correct number of iptable rules, err: %v", err)
+		// FIXME: This used to check that the no-snat rule had been hit, but nftables
+		// doesn't attach counters to rules unless you explicitly request them, which
```

Comment:
arent counters valuable for debugging? We have use them in the past to determine if our iptables rules are getting hit. Why wouldn't we enable them by default on nftables?

Reply:
even if we dont enable packet counters, wouldn't we still want to check that the nftable rule exists?

Reply:
There's no "enable counters by default" in nftables; they decided that had been a bad idea in iptables, since counting accurately requires some CPU synchronization so it slows everything down a little for something that you _usually_ don't care about.

But you can _explicitly_ add a counter to any rule where you want one.

```
add rule inet ovn-kubernetes mgmtport-snat oifname != "${MGMT_PORT_NAME}" return
add rule inet ovn-kubernetes mgmtport-snat meta l4proto . th dport @mgmtport-no-snat-nodeports return
add rule inet ovn-kubernetes mgmtport-snat ip daddr . meta l4proto . th dport @mgmtport-no-snat-services-v4 return
add rule inet ovn-kubernetes mgmtport-snat snat ip to ${MGMT_PORT_IP}
```

Should I add counters to the last three rules? (So then all packets destined for ovn-k8s-mp0 would get counted as one of "not SNATted because NodePort", "not SNATted because eTP:Local" or "SNATted".)

> even if we dont enable packet counters, wouldn't we still want to check that the nftable rule exists?

The e2e tests as a whole confirm that SNAT happens when it's supposed to happen and doesn't happen when it's not supposed to happen, so that implies that the rules are correct. (Also, the unit tests check that the rule exists.)

Reply:
we discussed on slack not including counters by default, but potentially adding them if config->loglevel is set to debug. I think besides that and a rebase this PR lgtm

Reply:
The individual unit tests you can change the log level if you want to. We run all our unit tests serially, and test cases should call "config.PrepareTestConfig()", which by default sets it to 5.

For e2e it probably just uses the kind.sh default:

```
    echo "-ml  | --master-loglevel            Log level for ovnkube (master), DEFAULT: 5."
    echo "-nl  | --node-loglevel              Log level for ovnkube (node), DEFAULT: 5"

```
Looks like from a job artifact log I can see:
```
2024-11-13T16:22:53.993355266Z stderr F I1113 16:22:53.993290    2511 config.go:2372] Logging config: {File:/var/log/ovn-kubernetes/ovnkube.log CNIFile:/var/log/ovn-kubernetes/ovn-k8s-cni-overlay.log LibovsdbFile: Level:5 LogFileMaxSize:100 LogFileMaxBackups:5 LogFileMaxAge:5 ACLLoggingRateLimit:20}

```